synapse/app/homeserver.py

```diff
@@ -241,16 +241,28 @@ def _configure_named_resource(self, name, compress=False):
             resources[SERVER_KEY_V2_PREFIX] = KeyApiV2Resource(self)
 
         if name == "webclient":
-            webclient_path = self.get_config().web_client_location
+            webclient_loc = self.get_config().web_client_location
 
-            if webclient_path is None:
+            if webclient_loc is None:
                 logger.warning(
                     "Not enabling webclient resource, as web_client_location is unset."
                 )
+            elif webclient_loc.startswith("http://") or webclient_loc.startswith(
+                "https://"
+            ):
+                resources[WEB_CLIENT_PREFIX] = RootRedirect(webclient_loc)
             else:
+                logger.warning(
+                    "Running webclient on the same domain is not recommended: "
+                    "https://github.com/matrix-org/synapse#security-note"
+                )
+                logger.warning(
+                    "After you move webclient to different host you can set "
+                    "web_client_location to its full URL to enable redirection."
+                )
```

Comment:
We should probably make this a single warning instead of two.

Reply:
Merged them, PTAL.